androidx.test.espresso.base.AssertionErrorHandler
    public transient void <init>(java.lang.Class[])
    {
        androidx.test.espresso.base.AssertionErrorHandler r0;
        java.lang.Class[] r1;

        r0 := @this: androidx.test.espresso.base.AssertionErrorHandler;

        r1 := @parameter0: java.lang.Class[];

        specialinvoke r0.<androidx.test.espresso.base.DefaultFailureHandler$TypedFailureHandler: void <init>(java.lang.Class[])>(r1);

        return;
    }

    public volatile void handleSafely(java.lang.Object, org.hamcrest.Matcher)
    {
        androidx.test.espresso.base.AssertionErrorHandler r0;
        java.lang.Object r1;
        org.hamcrest.Matcher r2;
        java.lang.Throwable $r3;

        r0 := @this: androidx.test.espresso.base.AssertionErrorHandler;

        r1 := @parameter0: java.lang.Object;

        r2 := @parameter1: org.hamcrest.Matcher;

        $r3 = (java.lang.Throwable) r1;

        virtualinvoke r0.<androidx.test.espresso.base.AssertionErrorHandler: void handleSafely(java.lang.Throwable,org.hamcrest.Matcher)>($r3, r2);

        return;
    }

    public void handleSafely(java.lang.Throwable, org.hamcrest.Matcher)
    {
        java.lang.Throwable r0;
        org.hamcrest.Matcher r1;
        java.lang.String $r2;
        java.lang.Thread $r3;
        java.lang.StackTraceElement[] $r4;
        androidx.test.espresso.base.AssertionErrorHandler r5;
        androidx.test.espresso.base.AssertionErrorHandler$AssertionFailedWithCauseError r6, r7;
        java.lang.Error r8;

        r5 := @this: androidx.test.espresso.base.AssertionErrorHandler;

        r0 := @parameter0: java.lang.Throwable;

        r1 := @parameter1: org.hamcrest.Matcher;

        r7 = new androidx.test.espresso.base.AssertionErrorHandler$AssertionFailedWithCauseError;

        r6 = r7;

        $r2 = virtualinvoke r0.<java.lang.Throwable: java.lang.String getMessage()>();

        specialinvoke r7.<androidx.test.espresso.base.AssertionErrorHandler$AssertionFailedWithCauseError: void <init>(java.lang.String,java.lang.Throwable)>($r2, r0);

        $r3 = staticinvoke <java.lang.Thread: java.lang.Thread currentThread()>();

        $r4 = virtualinvoke $r3.<java.lang.Thread: java.lang.StackTraceElement[] getStackTrace()>();

        r8 = (java.lang.Error) r6;

        virtualinvoke r8.<java.lang.Error: void setStackTrace(java.lang.StackTraceElement[])>($r4);

        throw r6;
    }

androidx.test.espresso.base.AssertionErrorHandler$AssertionFailedWithCauseError
    public void <init>(java.lang.String, java.lang.Throwable)
    {
        androidx.test.espresso.base.AssertionErrorHandler$AssertionFailedWithCauseError r0;
        java.lang.String r1;
        java.lang.Throwable r2;

        r0 := @this: androidx.test.espresso.base.AssertionErrorHandler$AssertionFailedWithCauseError;

        r1 := @parameter0: java.lang.String;

        r2 := @parameter1: java.lang.Throwable;

        specialinvoke r0.<junit.framework.AssertionFailedError: void <init>(java.lang.String)>(r1);

        virtualinvoke r0.<androidx.test.espresso.base.AssertionErrorHandler$AssertionFailedWithCauseError: java.lang.Throwable initCause(java.lang.Throwable)>(r2);

        return;
    }

